Namespace(server_address='192.168.0.17:8080', rounds=1, sample_fraction=1.0, min_num_clients=4)
The IP address of the server is: 192.168.0.17
The IP address of the client is: 192.168.0.14, 192.168.0.22, 192.168.0.15, 192.168.0.19
The ID of the client is: pi


 python3.10 server_dataset.py --server_address=192.168.0.17:8080 --rounds=1 --min_num_clients=4 | tee terminal-result_2024-06-20_mnist.txt
/usr/lib/python3/dist-packages/urllib3/util/ssl_.py:256: DeprecationWarning: ssl.PROTOCOL_TLS is deprecated
  context = SSLContext(ssl_version or PROTOCOL_TLS)
INFO :      Starting Flower server, config: num_rounds=1, no round_timeout
INFO :      Flower ECE: gRPC server running (1 rounds), SSL is disabled
INFO :      [INIT]
INFO :      Requesting initial parameters from one random client
INFO :      Received initial parameters from one random client
INFO :      Evaluating initial global parameters
INFO :
INFO :      [ROUND 1]
INFO :      configure_fit: strategy sampled 4 clients (out of 4)
INFO :      aggregate_fit: received 4 results and 0 failures
WARNING :   No fit_metrics_aggregation_fn provided
INFO :      configure_evaluate: strategy sampled 4 clients (out of 4)
INFO :      aggregate_evaluate: received 4 results and 0 failures
INFO :
INFO :      [SUMMARY]
INFO :      Run finished 1 rounds in 265.59s
INFO :      History (loss, distributed):
INFO :          '\tround 1: 0.5597280859947205\n'History (metrics, distributed, evaluate):
INFO :          {'accuracy': [(1, 0.91875)]}
INFO :
2024-06-20 16:41:13,638 - INFO - Start FL server.
Namespace(server_address='192.168.0.17:8080', rounds=1, sample_fraction=1.0, min_num_clients=4)
The IP address of the server is: 192.168.0.17
The IP address of the client is: 192.168.0.14, 192.168.0.22, 192.168.0.15, 192.168.0.19
The ID of the client is: pi

[rpi3b]
pi@raspberrypi3:~/E2FL$ python3.10 client_dataset.py --server_address=192.168.0.17:8080 --cid=0 --dataset=mnist
/home/pi/.local/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension:
  warn(f"Failed to load image Python extension: {e}")
Namespace(server_address='192.168.0.17:8080', cid=0, interface='wlan0', dataset='mnist', Model='ResNet20')
/home/pi/.local/lib/python3.10/site-packages/datasets/load.py:1461: FutureWarning: The repository for mnist contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/mnist
You can avoid this message in future by passing the argument `trust_remote_code=True`.
Passing `trust_remote_code=True` will be mandatory to load this dataset from the next major release of `datasets`.
  warnings.warn(
Downloading builder script: 100%|███████████| 3.98k/3.98k [00:00<00:00, 340kB/s]
Downloading readme: 100%|███████████████████| 6.83k/6.83k [00:00<00:00, 566kB/s]
Downloading data: 100%|████████████████████| 9.91M/9.91M [00:03<00:00, 2.85MB/s]
Downloading data: 100%|█████████████████████| 28.9k/28.9k [00:00<00:00, 586kB/s]
Downloading data: 100%|████████████████████| 1.65M/1.65M [00:00<00:00, 2.01MB/s]
Downloading data: 100%|█████████████████████| 4.54k/4.54k [00:00<00:00, 345kB/s]
Generating train split: 100%|█████| 60000/60000 [02:08<00:00, 465.87 examples/s]
Generating test split: 100%|██████| 10000/10000 [00:21<00:00, 470.67 examples/s]
2024-06-20 16:40:17,533 - INFO - ['Wi-Fi start: 1718869217.531287']
INFO flwr 2024-06-20 16:40:18,115 | grpc.py:52 | Opened insecure gRPC connection (no certificates were passed)
DEBUG flwr 2024-06-20 16:40:18,157 | connection.py:55 | ChannelConnectivity.IDLE
DEBUG flwr 2024-06-20 16:40:18,173 | connection.py:55 | ChannelConnectivity.CONNECTING
DEBUG flwr 2024-06-20 16:40:18,203 | connection.py:55 | ChannelConnectivity.READY
Client sampled for fit()
100%|███████████████████████████████████████████| 68/68 [00:09<00:00,  6.98it/s]
100%|███████████████████████████████████████████| 68/68 [00:08<00:00,  7.57it/s]
100%|███████████████████████████████████████████| 68/68 [00:08<00:00,  7.65it/s]
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:00<00:00,  3.68it/s]
DEBUG flwr 2024-06-20 16:41:13,646 | connection.py:220 | gRPC channel closed
INFO flwr 2024-06-20 16:41:13,647 | app.py:398 | Disconnect and shut down
2024-06-20 16:41:13,669 - INFO - ['Wi-Fi end: 1718869273.6678817']
2024-06-20 16:41:13,672 - INFO - The measurement data is saved as data_0_2024-06-20_16-41-13.pickle.



[rpi3b_2]
(env) pi@raspberrypi:~/E2FL $ python3.10 client_dataset.py --server_address=192.168.0.17:8080 --cid=1 --dataset=mnist
/home/pi/.local/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension:
  warn(f"Failed to load image Python extension: {e}")
Namespace(server_address='192.168.0.17:8080', cid=1, interface='wlan0', dataset='mnist', Model='ResNet20')
The repository for mnist contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/mnist.
You can avoid this prompt in future by passing the argument `trust_remote_code=True`.

Do you wish to run the custom code? [y/N] y
Downloading data: 100%|████████████████████| 9.91M/9.91M [00:04<00:00, 2.38MB/s]
Downloading data: 100%|█████████████████████| 28.9k/28.9k [00:00<00:00, 729kB/s]
Downloading data: 100%|████████████████████| 1.65M/1.65M [00:00<00:00, 1.99MB/s]
Downloading data: 100%|████████████████████| 4.54k/4.54k [00:00<00:00, 1.04MB/s]
Generating train split: 100%|█████| 60000/60000 [02:04<00:00, 483.54 examples/s]
Generating test split: 100%|██████| 10000/10000 [00:19<00:00, 506.68 examples/s]
2024-06-20 16:40:44,364 - INFO - ['Wi-Fi start: 1718869244.3624249']
INFO :
INFO :      Received: train message 3e364a10-ac87-4680-87f2-8c7dc50eba7e
Client sampled for fit()
100%|███████████████████████████████████████████| 68/68 [00:09<00:00,  7.22it/s]
100%|███████████████████████████████████████████| 68/68 [00:08<00:00,  7.72it/s]
100%|███████████████████████████████████████████| 68/68 [00:08<00:00,  7.67it/s]
INFO :      Sent reply
INFO :
INFO :      Received: evaluate message c2c7e463-4a85-4350-bd23-b84c3dc94d21
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:00<00:00,  4.05it/s]
INFO :      Sent reply
INFO :
INFO :      Received: reconnect message e0bca428-65fb-4b4c-bb9d-c2b75813b1c1
INFO :      Disconnect and shut down
2024-06-20 16:41:13,670 - INFO - ['Wi-Fi end: 1718869273.6681402']
2024-06-20 16:41:13,683 - INFO - The measurement data is saved as data_1_2024-06-20_16-41-13.pickle.



[rpi4]
Namespace(server_address='192.168.0.17:8080', cid=2, interface='wlan0', dataset='mnist', Model='ResNet20')
/home/pi/.local/lib/python3.10/site-packages/datasets/load.py:1461: FutureWarning: The repository for mnist contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/mnist
You can avoid this message in future by passing the argument `trust_remote_code=True`.
Passing `trust_remote_code=True` will be mandatory to load this dataset from the next major release of `datasets`.
  warnings.warn(
2024-06-20 16:36:58,443 - INFO - ['Wi-Fi start: 1718869018.4432497']
INFO flwr 2024-06-20 16:36:58,633 | grpc.py:52 | Opened insecure gRPC connection (no certificates were passed)
DEBUG flwr 2024-06-20 16:36:58,639 | connection.py:55 | ChannelConnectivity.IDLE
DEBUG flwr 2024-06-20 16:36:58,641 | connection.py:55 | ChannelConnectivity.CONNECTING
DEBUG flwr 2024-06-20 16:36:58,642 | connection.py:55 | ChannelConnectivity.READY
Client sampled for fit()
100%|███████████████████████████████████████████| 68/68 [00:02<00:00, 23.23it/s]
100%|███████████████████████████████████████████| 68/68 [00:02<00:00, 24.05it/s]
100%|███████████████████████████████████████████| 68/68 [00:02<00:00, 25.03it/s]
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:00<00:00, 15.77it/s]
DEBUG flwr 2024-06-20 16:41:13,644 | connection.py:220 | gRPC channel closed
INFO flwr 2024-06-20 16:41:13,645 | app.py:398 | Disconnect and shut down
2024-06-20 16:41:13,657 - INFO - ['Wi-Fi end: 1718869273.656924']
2024-06-20 16:41:13,658 - INFO - The measurement data is saved as data_2_2024-06-20_16-41-13.pickle.





[rpi5B]
Namespace(server_address='192.168.0.17:8080', cid=3, interface='wlan1', dataset='mnist', Model='ResNet20')
2024-06-20 16:36:47,640 - INFO - ['Wi-Fi start: 1718869007.640533']
INFO :
INFO :      Received: get_parameters message 30f91a06-2160-4308-85ac-0d7530fb9905
INFO :      Sent reply
INFO :
INFO :      Received: train message e8896dad-f8f0-44ba-a92f-a00f83f7a4ac
Client sampled for fit()
100%|███████████████████████████████████████████| 68/68 [00:01<00:00, 51.46it/s]
100%|███████████████████████████████████████████| 68/68 [00:01<00:00, 56.96it/s]
100%|███████████████████████████████████████████| 68/68 [00:01<00:00, 58.05it/s]
INFO :      Sent reply
INFO :
INFO :      Received: evaluate message 3023b86f-e643-4ef4-81cb-e7c3ef90070c
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:00<00:00, 26.16it/s]
INFO :      Sent reply
INFO :
INFO :      Received: reconnect message cc63643b-66a0-4e0e-a2ac-e64053a7a926
INFO :      Disconnect and shut down
2024-06-20 16:41:13,648 - INFO - ['Wi-Fi end: 1718869273.6477966']
2024-06-20 16:41:13,648 - INFO - The measurement data is saved as data_3_2024-06-20_16-41-13.pickle.


